orderprocessor/src/main/java/com/homeaway/streamplatform/hellostreams/orderprocessor/dao/CustomerDao.java

```diff
@@ -5,25 +5,42 @@
 import org.springframework.stereotype.Repository;
 
 import java.time.ZonedDateTime;
+import java.util.HashMap;
 import java.util.Map;
 
 @Repository
 public class CustomerDao {
     public static final String CUSTOMER_TEST_ID = "1819916e-0fd2-411b-9832-a0eb62348c17";
-    private static final Map<String,Customer> TEST_CUSTOMER_CACHE = Map.of(CUSTOMER_TEST_ID, createTestCustomer());
+
+    private static final Map<String,Customer> TEST_CUSTOMER_CACHE = new HashMap<>(){
+        {
+            put(CUSTOMER_TEST_ID, createTestCustomer(CUSTOMER_TEST_ID, "neoward")); // seed the database with a user
+        }
+    };
+
 
     public Customer getCustomer(String customerId) {
+        Boolean customerMissing = !TEST_CUSTOMER_CACHE.containsKey(customerId);
+        if (customerMissing) {
+            String username = customerId;
+            insertCustomer(customerId, username); // if the customer isn't in the database yet add them
+        }
```

Comment:
Business logic SHOULD NOT be in Dao.
Please move to service-layer -->CustomerService